onnxruntime/core/platform/windows/env.cc

```diff
@@ -202,643 +226,610 @@ class WindowsThread : public EnvThread {
+  // Therefore, If the function succeeds, kBufferSize should be larger than char_count.
+  auto char_count = GetEnvironmentVariableA(var_name.c_str(), buffer.data(), kBufferSize);
 
-    if (kBufferSize > char_count) {
-      buffer.resize(char_count);
-      return buffer;
-    }
+  if (kBufferSize > char_count) {
+    buffer.resize(char_count);
+    return buffer;
+  }
 
-    // Else either the call was failed, or the buffer wasn't large enough.
-    // TODO: Understand the reason for failure by calling GetLastError().
-    // If it is due to the specified environment variable being found in the environment block,
-    // GetLastError() returns ERROR_ENVVAR_NOT_FOUND.
-    // For now, we assume that the environment variable is not found.
+  // Else either the call was failed, or the buffer wasn't large enough.
+  // TODO: Understand the reason for failure by calling GetLastError().
+  // If it is due to the specified environment variable being found in the environment block,
+  // GetLastError() returns ERROR_ENVVAR_NOT_FOUND.
+  // For now, we assume that the environment variable is not found.
 
-    return std::string();
-  }
+  return std::string();
+}
 
- private:
-  WindowsEnv() = default;
+ProcessorInfo WindowsEnv::GetProcessorAffinityMask(int global_processor_id) const {
+  if (global_processor_info_map_.count(global_processor_id)) {
+    return global_processor_info_map_.at(global_processor_id);
+  } else {
+    return {-1, -1};
+  }
+}
 
-  typedef VOID(WINAPI* FnGetSystemTimePreciseAsFileTime)(LPFILETIME);
-  WindowsTelemetry telemetry_provider_;
-};
-}  // namespace
+WindowsEnv::WindowsEnv() {
+  InitializeCpuInfo();
+}
 
-Env& Env::Default() {
-  return WindowsEnv::Instance();
+/*
+Discover all cores in a windows system.
+Note - every "id" here, given it be group id, core id, or logical processor id, starts from 0.
+*/
+void WindowsEnv::InitializeCpuInfo() {
+  DWORD returnLength = 0;
+  GetLogicalProcessorInformationEx(RelationProcessorCore, nullptr, &returnLength);
+  auto last_error = GetLastError();
+  if (last_error != ERROR_INSUFFICIENT_BUFFER) {
+    return;
+  }
+
+  std::unique_ptr<char[]> allocation = std::make_unique<char[]>(returnLength);
+  SYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX* processorInfos = reinterpret_cast<SYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX*>(allocation.get());
+
+  if (!GetLogicalProcessorInformationEx(RelationProcessorCore, processorInfos, &returnLength)) {
+    return;
+  }
+
+  int global_processor_id = 0;
+  const BYTE* iter = reinterpret_cast<const BYTE*>(processorInfos);
+  const BYTE* end = iter + returnLength;
+  while (iter < end) {
+    auto processor_info = reinterpret_cast<const SYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX*>(iter);
+    auto size = processor_info->Size;
+    if (processor_info->Relationship == RelationProcessorCore &&
+        processor_info->Processor.GroupCount == 1) {
```

Comment:
>GroupCount == 1) [](http://example.com/codeflow?start=34&length=16)

Need a comment as to what 1 means in this case.

Reply:
Commented like "Discoverred a phyical core and it belongs exclusively to a single group".